src/vm/compiler.c

```diff
@@ -1445,7 +1445,8 @@ ParseRule rules[] = {
         {NULL,     NULL,      PREC_NONE},               // TOKEN_IF
         {NULL,     and_,      PREC_AND},                // TOKEN_AND
         {NULL,     NULL,      PREC_NONE},               // TOKEN_ELSE
-        {NULL,     or_,       PREC_OR},                 // TOKEN_OR
+        {NULL,     or_,       PREC_OR},                 // TOKEN_OR               
+        {NULL,	   NULL,      PREC_NONE},               //TOKEN_MATCH 
```

Comment:
The errors will be stemming from the fact you've added two new tokens, however, here you've only added one of the tokens to the array. The order of the enum is what is used to index on this array, so even though there's no logic required in here for `TOKEN_CASE` it still needs to be added here.

Hopefully that makes sense!

```suggestion
        {NULL,	   NULL,      PREC_NONE},               // TOKEN_MATCH 
        {NULL,	   NULL,      PREC_NONE},               // TOKEN_CASE
```